src/pool/srv_target.c

```diff
@@ -1060,13 +1060,15 @@ ds_pool_tgt_connect(struct ds_pool *pool, struct pool_iv_conn *pic)
 	ds_pool_get(pool);
 	hdl->sph_pool = pool;
 
-	cred_iov.iov_len = pic->pic_cred_size;
-	cred_iov.iov_buf_len = pic->pic_cred_size;
-	cred_iov.iov_buf = &pic->pic_creds[0];
-	rc = daos_iov_copy(&hdl->sph_cred, &cred_iov);
-	if (rc != 0) {
-		ds_pool_put(pool);
-		D_GOTO(out, rc);
+	if (pic->pic_cred_size) {
```

Comment:
@wangdi1, @dpquigl, do we expect any cases where pic_cred_size is 0? I'm asking because it _seems_ if hdl->sph_cred is left in "all zero" state, it may trigger errors or problems when other places try to use it.

Reply:
Yup, this is what i am not sure in which case pic_cred_size will be 0, but i indeed see this corruptions on one of customer site.

Reply:
yes, cred_iov comes from sec capability get in ds_pool_connect_handler(), it could be zero I assume.

Reply:
The length of the cred should never be zero. I'd imagine this is a bug. The credential is checked in the top level pool connect handler though on the pool service leader. I don't believe it gets checked again in the target connect cases.